openmetadata-ui/src/main/resources/ui/src/pages/DataModelPage/DataModelPage.component.tsx

```diff
@@ -568,162 +466,33 @@ const DataModelsPage = () => {
-            <Tabs.TabPane
-              key={DATA_MODELS_DETAILS_TABS.SQL}
-              tab={
-                <span data-testid={DATA_MODELS_DETAILS_TABS.SQL}>
-                  {t('label.sql-uppercase')}
-                </span>
-              }>
-              <Card className="h-full">
-                <SchemaEditor
-                  editorClass="custom-code-mirror-theme full-screen-editor-height"
-                  mode={{ name: CSMode.SQL }}
-                  options={{
-                    styleActiveLine: false,
-                    readOnly: 'nocursor',
-                  }}
-                  value={dataModelData.sql}
-                />
-              </Card>
-            </Tabs.TabPane>
-          )}
-
-          <Tabs.TabPane
-            key={DATA_MODELS_DETAILS_TABS.LINEAGE}
-            tab={
-              <span data-testid={DATA_MODELS_DETAILS_TABS.LINEAGE}>
-                {t('label.lineage')}
-              </span>
-            }>
-            <Card
-              className="h-full card-body-full"
-              data-testid="lineage-details">
-              <EntityLineageComponent
-                deleted={deleted}
-                entityType={EntityType.DASHBOARD_DATA_MODEL}
-                hasEditAccess={hasEditLineagePermission}
-              />
-            </Card>
-          </Tabs.TabPane>
-        </Tabs>
-
-        {threadLink ? (
-          <ActivityThreadPanel
-            createThread={createThread}
-            deletePostHandler={deletePostHandler}
-            open={Boolean(threadLink)}
-            postFeedHandler={postFeedHandler}
-            threadLink={threadLink}
-            updateThreadHandler={updateThreadHandler}
-            onCancel={onThreadPanelClose}
-          />
-        ) : null}
-      </div>
-    </PageContainerV1>
+    <DataModelDetails
+      activeTab={tab}
+      createThread={createThread}
+      dashboardDataModelFQN={dashboardDataModelFQN}
+      dataModelData={dataModelData}
+      dataModelPermissions={dataModelPermissions}
+      deletePostHandler={deletePostHandler}
+      entityFieldTaskCount={entityFieldTaskCount}
+      entityFieldThreadCount={entityFieldThreadCount}
+      entityThread={entityThread}
+      feedCount={feedCount}
+      fetchFeedHandler={getFeedData}
+      handleFeedFilterChange={handleFeedFilterChange}
+      handleFollowDataModel={handleFollowDataModel}
+      handleRemoveTier={handleRemoveTier}
+      handleTabChange={handleTabChange}
+      handleUpdateDataModel={handleUpdateDataModel}
+      handleUpdateDescription={handleUpdateDescription}
+      handleUpdateOwner={handleUpdateOwner}
+      handleUpdateTags={handleUpdateTags}
+      handleUpdateTier={handleUpdateTier}
+      isEntityThreadLoading={isEntityThreadLoading}
+      paging={paging}
+      postFeedHandler={postFeedHandler}
+      updateThreadHandler={updateThreadHandler}
+    />
   );
 };
 
-export default DataModelsPage;
+export default observer(DataModelsPage);
```

Comment:
Why we need observer here? Due to AppState?

Reply:
yes, we are getting user details from it.